x-pack/filebeat/input/awscloudwatch/cloudwatch.go

```diff
@@ -44,7 +46,8 @@ type workResponse struct {
 func newCloudwatchPoller(log *logp.Logger, metrics *inputMetrics,
 	awsRegion string, config config, stateHandler *stateHandler) *cloudwatchPoller {
 	if metrics == nil {
-		metrics = newInputMetrics("", nil)
+		metrics = newInputMetrics(
+			v2.Context{MetricsRegistry: monitoring.NewRegistry()})
```

Comment:
This part confuses me, why are we creating a new registry with a faked context instead of just using the real `v2.Context`? The caller of `newCloudwatchPoller` has one that could be passed as a parameter instead.

Reply:
here we don't want a registry. Before passing an empty ID `newInputMetrics("", nil)` would not register metrics. I just kept the same behaviour with the new API.